### PR TITLE
Validator

### PR DESCRIPTION
**Pull request by j-mai**
This PR changes `EventValidator` and `EventValidation` into a more general `Validator` interface so that validators can be added and expanded upon more easily.

### CODE REVIEW

File: message/validation/validator.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"time"
+
+	"github.com/xmidt-org/glaukos/message"
+)
+
+var (
+	ErrInvalidEventType = errors.New("event type doesn't match")
+	ErrInvalidBootTime  = errors.New("invalid boot-time")
+	ErrInvalidBirthdate = errors.New("invalid birthdate")
+	ErrTimeParse        = errors.New("parsing error")
```
Comment (kristinaspring) [inline message/validation/validator.go:15]: Are these three errors still needed?

File: message/validation/timeValidator.go [hunk 2]
```diff
@@ -2,31 +2,29 @@ package validation
...
 	ErrPastDate    = errors.New("date is too far in the past")
-	ErrNilTimeFunc = errors.New("currenttime function has not been set")
+	ErrNilTimeFunc = errors.New("current-time function has not been set")
 )
 
 // TimeValidation sees if a given time is within the time frame it is set to validate
 type TimeValidation interface {
-	IsTimeValid(time.Time) (bool, error)
-	CurrentTime() time.Time
+	ValidTime(time.Time) (bool, error)
```
Comment (kristinaspring) [inline message/validation/timeValidator.go:16]: I don't think you need `Time` in the function name, but I don't feel that strongly about this.  If you like it, then keep it. 👍

**Review by kristinaspring — APPROVED** [review]
Looks good! Two small comments.

File: message/validation/timeLocation.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	Boottime
+)
+
+var (
+	timeLocationUnmarshal = map[string]TimeLocation{
+		"birthdate": Birthdate,
+		"boot-time": Boottime,
+	}
+)
+
```
Comment (kristinaspring) [inline message/validation/timeLocation.go:23]: Where are these two functions being used? Or is this prep for future work?

File: message/validation/validator.go (hunk 1 above)
Comment (j-mai) [reply message/validation/validator.go]: Nope! Deleted

File: message/validation/timeLocation.go (hunk 3 above)
Comment (j-mai) [reply message/validation/timeLocation.go:23]: This is prep for future work

**Pull request merged**